### DISCUSSION TITLE
Allow "Accept" and Reject" in set-cookie

### DISCUSSION THREAD

**Discussion by ryanbr** [uBlockOrigin/uAssets#18762] [General] [answered]
Allow "Accept" and Reject" in set-cookie, similar to true/false

As seen on `https://www.calvinklein.co.uk/`

```
calvinklein.co.uk##+js(set-cookie, PVH_COOKIES_GDPR, Accept)
calvinklein.co.uk##+js(set-cookie, PVH_COOKIES_GDPR_SOCIALMEDIA, Reject)
calvinklein.co.uk##+js(set-cookie, PVH_COOKIES_GDPR_ANALYTICS, Reject)
```

**Comment by gorhill** [comment, marked as answer]
https://github.com/gorhill/uBlock/commit/1f57822c337791202a66374b37887cb7b8d6697f

**Reply by peace2000** [reply]
If we allow `Accept` and `Reject` in capitalized form, why not add minuscule versions as well?

-> `accept` and `reject`

@gorhill

**Reply by iam-py-test** [reply]
Also maybe all-caps (no idea if those are used in the real world, though)

**Reply by krystian3w** [reply]
One could also extend "local/session-storage" as there is no proven malware/trackers what uses `Accept`, `Reject`, `OK`, `Y`, `y`, `N`, `n`.

> https://github.com/gorhill/uBlock/blob/1f57822c337791202a66374b37887cb7b8d6697f/assets/resources/scriptlets.js#L2818

> https://github.com/gorhill/uBlock/blob/9d20cbe6afb3f06a7efa6d3e5455c836762da563/assets/resources/scriptlets.js#L2926

> https://github.com/gorhill/uBlock/blob/9d20cbe6afb3f06a7efa6d3e5455c836762da563/assets/resources/scriptlets.js#L759-L778

**Reply by peace2000** [reply]
Or would it be possible to make the scriptlet case-insensitive concerning its predefined values?

E.G. the scriptlet would accept `true`, `TRUE`, `True`, `tRuE` and all the possible versions.

Though if this requires a lot of coding, might not be worth it, but if there were a trivial change that can be made...

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/03d0d8d4ce7ac64434266e25b0ff402245720f65

**Comment by Yuki2718** [comment]
https://github.com/easylist/easylist/issues/16760 `allow` may be worth adding. They don't have `deny` value, but maybe should also be added?

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/03d0d8d4ce7ac64434266e25b0ff402245720f65

**Reply by peace2000** [reply]
Adguard added `deny` value as well: https://github.com/AdguardTeam/Scriptlets/commit/6f36f888bb4a1ede436f5bbe2d2d3af28e05ff8d

**Reply by krystian3w** [reply]
Add after bump library in all environment.

**Comment by peace2000** [comment]
> Adguard added `deny` value as well: [AdguardTeam/Scriptlets@6f36f88](https://github.com/AdguardTeam/Scriptlets/commit/6f36f888bb4a1ede436f5bbe2d2d3af28e05ff8d)

https://github.com/gorhill/uBlock/commit/da8fe2f950da7d4232840822d994c053f6807712
